mitodl/mit-open
# Add a user menu with login, logout to UI

# What are the relevant tickets?
Closes #241 

# Description (What does it do?)
This PR adds a user menu in top-right corner with log in / log out links as appropriate. A few other links are added, too.

**Authenticated user / Anonymous user** *Note, "Learning Paths" only available to those with editing permissions".
    
<img width="225" alt="Screenshot 2023-12-19 at 9 57 14 AM" src="https://github.com/mitodl/mit-open/assets/9010790/073dfc17-7c2b-4aea-905d-e20985361ddd"> / <img width="200" alt="Screenshot 2023-12-19 at 9 57 23 AM" src="https://github.com/mitodl/mit-open/assets/9010790/30c8207b-d7c7-4b69-9f0d-81fc11813e3d">

Additionally, I made some updates to the reusable SimpleMenu component.

# How can this be tested?

**Prerequisites:** Connect your local instance to keycloak; either local keycloak or on RC. Ask me for creds if you need them.

Assuming you are initially unauthenticated:

1. **Unauthenticated:** Visit http://localhost:8063/ ; in top right, click user menu and log in. You should be redirected to keycloak, after which you're redirected back to MIT Open.
2. **Authenticated:** Now click usermenu
    - if your user has `is_learning_path_editor` permission, they'll see a "Learning Paths" link.
    - all authenticated users see "Log out" link.
3. **Authenticated:** Try logging out. It should work.
    - Note: Does not currently redirect to MIT Open


**Some accessibility stuff:** This is most easily tested as a superuser, since you'll have multiple links int he dropdown usermenu.
- Check that you can tab to the usermenu trigger and open it with keyboard
- Once it's open, check that navigation with keyboard arrows works.
    - The indidividual menu items are navigable with keyboard arrows, NOT via tab. See https://www.w3.org/WAI/ARIA/apg/patterns/menubar/
- The whole menu item should be clickable as a link
    <img width="310" alt="Screenshot 2023-12-19 at 10 12 02 AM" src="https://github.com/mitodl/mit-open/assets/9010790/557997d5-4f8c-4160-a37f-d1b56cc4d039">
